pkg/operator/controller/externaldns/controller_test.go

```diff
@@ -128,12 +188,44 @@ func TestReconcile(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			cl := fake.NewClientBuilder().WithScheme(test.Scheme).WithRuntimeObjects(tc.existingObjects...).Build()
+			kubeFakeClient := kubernetesFakeClient.NewSimpleClientset()
+
 			r := &reconciler{
-				client: cl,
-				scheme: test.Scheme,
-				config: tc.inputConfig,
-				log:    zap.New(zap.UseDevMode(true)),
+				client:     cl,
+				scheme:     test.Scheme,
+				config:     tc.inputConfig,
+				log:        zap.New(zap.UseDevMode(true)),
+				kubeclient: kubeFakeClient.Discovery(),
+			}
+
+			var infraConfig configv1.Infrastructure
+
+			if tc.ocpPlatform {
+				kubeFakeClient.Fake.Resources = getfakeResource()
+				operatorv1alpha1.IsOpenShift = true
+				infraConfig = configv1.Infrastructure{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Infrastructure",
+						APIVersion: "config.openshift.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "cluster",
+					},
+					Status: configv1.InfrastructureStatus{
+						PlatformStatus: &configv1.PlatformStatus{
+							Type: configv1.GCPPlatformType,
+						},
+					},
+				}
```

Comment:
why this is simply not added to the testcase's `existingObjects` field?

Reply:
remove it as it is not needed